### PR TITLE
refactor nested ifs and add functions for checks

### PR DESCRIPTION
**Pull request by thoresuenert**
## Description

Sanitise the existing code by making use of early return instead of nesting ifs. 
Introduce functions for relevant security checks to be explicit what the check is about. 
I replaced the existing logic but maybe the admin and logged in checks are not dependent (dont know yet how those annotations can be used).
Then it would be possible to split checkUserPermission to checkUserIsLoggedIn and checkUserIsAdmin.

cc @cdamken 💋 

## Motivation and Context

increase readability of code

## How Has This Been Tested?

Hopefully fully tested before change

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: lib/private/AppFramework/Middleware/Security/SecurityMiddleware.php [hunk 1]
```diff
@@ -185,33 +170,74 @@ public function afterController($controller, $methodName, Response $response) {
...
+		$response = new TemplateResponse('core', '403', ['file' => $exception->getMessage()], 'guest');
+		$response->setStatus($exception->getCode());
+		return $response;
 	}
+	
+	/**
+	* Check if user has permission to access controller method
+	* 
+	* @throws NotLoggedInException when user has to be logged in and isnt
+	* @throws NotAdminException when user has to be logged in and isnt
```
Comment (phil-davis) [inline lib/private/AppFramework/Middleware/Security/SecurityMiddleware.php]: NotAdminException when user has to be admin and isnt

**Comment by DeepDiver1975** [comment]
Honestly I  disagree with changes which shift code around just for the sake of making-in-nicer

Furthermore I see no real issue with loops and ifs.
One might use array_* functions for a more functional coding style.

**Comment by thoresuenert** [comment]
@DeepDiver1975 thanks for your fast and honest reply.
Both PRs(#28241) were a test for me if my philosophy to code fits the project.

**Comment by patrickjahns** [comment]
Sad that this was never picked up 

> Honestly I disagree with changes which shift code around just for the sake of making-in-nicer

Very much disagree - [and the internet does so too](http://lmgtfy.com/?q=code+is+more+often+read+than+written)

**Pull request closed without merging**